lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -1135,35 +1081,13 @@ void ReabstractionInfo::specializeConcreteAndGenericSubstitutions(
     SpecializedGenericEnv = GenPair.first;
   }
 
-  // Map callee's interface types to specialized function interface types.
-  auto CalleeInterfaceToSpecializedInterfaceType =
-      [&](SubstitutableType *CalleeInterfaceType) -> Type {
-    // First map the callee's interface type to the caller's contextual type.
-    auto Ty = CalleeInterfaceToCallerArchetypeMap.lookupSubstitution(
-        cast<SubstitutableType>(CalleeInterfaceType->getCanonicalType()));
-    assert(Ty);
-    // Then map the caller's contextual type to an interface type in the
-    // specialized generic environment.
-    auto SpecializedInterfaceTy = Ty.subst(
-        [&](SubstitutableType *archetype) -> Type {
-          auto Ty = CallerInterfaceToSpecializedInterfaceMapping.lookup(
-              CallerGenericEnv
-                  ->mapTypeOutOfContext(archetype->getCanonicalType())
-                  ->getCanonicalType());
-          assert(Ty);
-          return Ty;
-        },
-        LookUpConformanceInSignature(*SpecializedGenericSig));
-    return SpecializedInterfaceTy;
-  };
-
   {
     SmallVector<Substitution, 4> List;
 
     CalleeGenericSig->getSubstitutions(
```

Comment:
This should just be CalleeGenericSig->GetSubstitutions(CallerInterfaceToSpecializedInterfaceMap, List)